### PR TITLE
Force layout when vertical tab strip bounds changed

### PR DESCRIPTION
**Pull request by sangwoo108**
* When drag-and-drop session is active or finished, tab strip sometimes doesn't lay out itself. But when the parent view's bounds changes, we should force it to layout.

* In addition to that, this PR makes the vertical tab's state after drag and drop session. We collapsed it regardless mouse-hover state. But as of now, we'll set it 'floating' mode when mouse is over the tab strip.

Resolves https://github.com/brave/brave-browser/issues/29941
Resolves https://github.com/brave/brave-browser/issues/29937

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
* Enable vertical tab strip and floating mode
* Test 1: Press down a tab before the vertical tab strip is expanded and don't release the mouse
  * Tabs should be as wide as tab strip.
* Test 2: When dragging a tab into another window and drop it there, tab strip should stay open if mouse is over tab strip.

### CODE REVIEW

**Review by simonhong — COMMENTED** [review]
While testing this PR on my local, I'm seeing this backtrace.
```
[16168:26232:0425/142307.505:ERROR:vertical_tab_strip_region_view.cc(609)] ExpandTabStripForDragging Backtrace:
        base::debug::CollectStackTrace [0x00007FFBCC6C14E2+18] (C:\PRJ\brave-browser2\src\base\debug\stack_trace_win.cc:329)
        base::debug::StackTrace::StackTrace [0x00007FFBCC6AB73D+29] (C:\PRJ\brave-browser2\src\base\debug\stack_trace.cc:221)
        VerticalTabStripRegionView::ExpandTabStripForDragging [0x00007FFBABECCD6F+165] (C:\PRJ\brave-browser2\src\brave\browser\ui\views\frame\vertical_tab_strip_region_view.cc:609)
        TabDragController::DetachAndAttachToNewContext [0x00007FFBABEA8D50+208] (C:\PRJ\brave-browser2\src\brave\browser\ui\views\tabs\tab_drag_controller.cc:252)
        TabDragControllerChromium::RunMoveLoop [0x00007FFBAD095396+794] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_drag_controller.cc:1613)
        TabDragControllerChromium::DetachIntoNewBrowserAndRunMoveLoop [0x00007FFBAD097041+999] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_drag_controller.cc:1526)
        TabDragControllerChromium::DragBrowserToNewTabStrip [0x00007FFBAD0969E6+304] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_drag_controller.cc:929)
        TabDragControllerChromium::ContinueDragging [0x00007FFBAD095641+317] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_drag_controller.cc:890)
        TabDragControllerChromium::Drag [0x00007FFBAD0939E3+1301] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_drag_controller.cc:627)
        TabStrip::TabDragContextImpl::ContinueDrag [0x00007FFBAD0808A8+196] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_strip.cc:317)
        TabStrip::TabDragContextImpl::OnMouseDragged [0x00007FFBAD08355F+15] (C:\PRJ\brave-browser2\src\chrome\browser\ui\views\tabs\tab_strip.cc:183)
        views::View::ProcessMouseDragged [0x00007FFBAA6D2E84+356] (C:\PRJ\brave-browser2\src\ui\views\view.cc:3428)
        ui::EventDispatcher::DispatchEvent [0x00007FFBD4ED6576+118] (C:\PRJ\brave-browser2\src\ui\events\event_dispatcher.cc:188)
        ui::EventDispatcher::ProcessEvent [0x00007FFBD4ED61FB+315] (C:\PRJ\brave-browser2\src\ui\events\event_dispatcher.cc:137)
        ui::EventDispatcherDelegate::DispatchEventToTarget [0x00007FFBD4ED6010+96] (C:\PRJ\brave-browser2\src\ui\events\event_dispatcher.cc:83)
        ui::EventDispatcherDelegate::DispatchEvent [0x00007FFBD4ED5F0B+155] (C:\PRJ\brave-browser2\src\ui\events\event_dispatcher.cc:54)
        views::internal::RootView::OnMouseDragged [0x00007FFBAA6E3A2E+190] (C:\PRJ\brave-browser2\src\ui\views\widget\root_view.cc:510)
        views::Widget::OnMouseEvent [0x00007FFBAA6EEFDC+236] (C:\PRJ\brave-browser2\src\ui\views\widget\widget.cc:1751)
        views::NativeWidgetAura::OnMouseEvent [0x00007FFBAA72A385+197] (C:\PRJ\brave-browser2\src\ui\views\widget\native_widget_aura.cc:1121)
        ui::EventDispatcher::DispatchEvent [0x00007FFBD4ED6576+118] (C:\PRJ\brave-browser2\src\ui\events\event_dispatcher.cc:188)

```

File: browser/ui/views/frame/vertical_tab_strip_region_view.cc [hunk 1]
```diff
@@ -589,14 +589,30 @@ void VerticalTabStripRegionView::SetState(State state) {
...
+  }
+
+  SetState(State::kCollapsed);
+}
+
 VerticalTabStripRegionView::ScopedStateResetter
 VerticalTabStripRegionView::ExpandTabStripForDragging() {
-  if (state_ == State::kExpanded)
+  if (state_ == State::kExpanded) {
+    LOG(ERROR) << __FUNCTION__ << " " << base::debug::StackTrace(20);
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply browser/ui/views/frame/vertical_tab_strip_region_view.cc:609]: Ah, yeah I need to remove this.

**Review by simonhong — APPROVED** [review]

File: browser/ui/views/tabs/brave_tab_group_header.cc [hunk 2]
```diff
@@ -40,6 +43,25 @@ SkColor GetGroupBackgroundColorForVerticalTabs(
...
+                                ->widget_delegate()
+                                ->AsBubbleDialogDelegate();
+    DCHECK(bubble_delegate);
+    // Technically, this can happen when tab strip's orientation changes
+    // with the editor bubble open. It re-parents the widget which can cause
+    // DCHECK failure. We should call SetAnchorView again to reset
+    // the anchor widget.
+    bubble_delegate->SetAnchorView(this);
+  }
+}
```
Comment (sangwoo108) [inline browser/ui/views/tabs/brave_tab_group_header.cc:63]: @simonhong Do you mind take another look? This part was added to fix test failure on Linux.

**Review by simonhong — APPROVED** [review]
Still ++ 👍🏼

**Pull request merged**